test/widgets/profile_test.dart

```diff
@@ -0,0 +1,269 @@
+
+  group('ProfilePage', () {
+    testWidgets('page builds; profile page renders', (WidgetTester tester) async {
+      final user = eg.user(userId: 1, fullName: 'test user');
+
+      await setupPage(tester, users:[user], pageUserId: user.userId);
+
+      check(because: 'find user avatar', find.byType(Avatar).evaluate()).length.equals(1);
+      check(because: 'find user name', find.text('test user').evaluate()).isNotEmpty();
+    });
+
+    testWidgets('page builds; profile page renders with profileData', (WidgetTester tester) async {
+      await setupPage(tester,
+        users: [
+          eg.user(userId: 1, profileData: {
+            0: ProfileFieldUserData(value: 'shortTextValue'),
+            1: ProfileFieldUserData(value: 'longTextValue'),
+            2: ProfileFieldUserData(value: 'x'),
+            3: ProfileFieldUserData(value: 'dateValue'),
+            4: ProfileFieldUserData(value: 'http://example/linkValue'),
+            5: ProfileFieldUserData(value: '[2]'),
+            6: ProfileFieldUserData(value: 'externalValue'),
+            7: ProfileFieldUserData(value: 'pronounsValue'),
+          }),
+          eg.user(userId: 2),
+        ],
+        pageUserId: 1,
+        customProfileFields: [
+          mkCustomProfileField(0, CustomProfileFieldType.shortText),
+          mkCustomProfileField(1, CustomProfileFieldType.longText),
+          mkCustomProfileField(2, CustomProfileFieldType.choice, fieldData: '{"x": {"text": "choiceValue", "order": "1"}}'),
+          mkCustomProfileField(3, CustomProfileFieldType.date),
+          mkCustomProfileField(4, CustomProfileFieldType.link),
+          mkCustomProfileField(5, CustomProfileFieldType.user),
+          mkCustomProfileField(6, CustomProfileFieldType.externalAccount, fieldData: '{"subtype": "external1"}'),
+          mkCustomProfileField(7, CustomProfileFieldType.pronouns),
+        ], realmDefaultExternalAccounts: {
+          'external1': RealmDefaultExternalAccount(
+            name: 'external1',
+            text: '',
+            hint: '',
+            urlPattern: 'https://example/%(username)s')});
+
+      final testCases = [
+        (find.text('field0'), find.text('shortTextValue'), CustomProfileFieldType.shortText),
+        (find.text('field1'), find.text('longTextValue'), CustomProfileFieldType.longText),
+        (find.text('field2'), find.text('choiceValue'), CustomProfileFieldType.choice),
+        (find.text('field3'), find.text('dateValue'), CustomProfileFieldType.date),
+        (find.text('field4'), find.text('http://example/linkValue'), CustomProfileFieldType.link),
+        (find.text('field5'), find.byType(Avatar), CustomProfileFieldType.user),
+        (find.text('field6'), find.text('http://example/externalValue'), CustomProfileFieldType.externalAccount),
+        (find.text('field7'), find.text('pronounsValue'), CustomProfileFieldType.pronouns),
+      ];
+      final profileDataTable = find.byType(Column).last;
+      testCases.map((testCase) {
```

Comment:
Generally prefer direct for-loops for plain iteration:
```suggestion
      for (final testCase in testCases) {
```

… Oh huh, and that evokes a test failure:
```
$ time flutter test test/widgets/profile_test.dart
00:02 +1: ProfilePage page builds; profile page renders with profileData                                              
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: a int that:
  equals <1>
Actual: <0>
Which: are not equal
Reason: find label for CustomProfileFieldType.shortText
```

Ah, I see why that wasn't throwing. It's because `map` is *lazy* — it just returns an Iterable, not a materialized List — and that iterable wasn't getting iterated through, so the body was never getting invoked.

Reply:
I fixed this, but additionally removed the search for the profile fields as an ancestor to all the finders. The way to identify that private widget was quite brittle, and I don't think it was that important to ensure the found fields are embedded in the right widget, especially if the layout gets refactored in the future.

Reply:
Cool, that reasoning makes sense to me.